tests/modules/commandfilter.c

```diff
@@ -89,6 +89,37 @@ int CommandFilter_LogCommand(RedisModuleCtx *ctx, RedisModuleString **argv, int
     return REDISMODULE_OK;
 }
 
+static int blmove_count = 0;
+
+/* Filter to protect against Bug #11894 reappearing
+ *
+ * ensures that the filter is only run the first time through, and not on reprocessing
+ */
+void CommandFilter_Blmove(RedisModuleCommandFilterCtx *filter)
+{
+    if (RedisModule_CommandFilterArgsCount(filter) != 6)
+        return;
+
+    RedisModuleString *arg = RedisModule_CommandFilterArgGet(filter, 0);
+    size_t arg_len;
+    const char *arg_str = RedisModule_StringPtrLen(arg, &arg_len);
+
+    if (arg_len != 6 || strncmp(arg_str, "blmove", 6))
+        return;
+
+
+    if (blmove_count == 0) {
+        blmove_count++;
+        return;
+    }
```

Comment:
hold on.. maybe i don't understand.
why do we need this? i thought the idea was that we'll let the command filter run as many times as it wants, and we know how many times it run by monitoring the effect (if LEFT and RIGHT are swapped).

if i now see that the filter exists on the first time it runs, then how come the test got the argument swapped? or maybe i'm wrong?

Reply:
the point of the filter is to not run (i.e. it just increments the counter past 0) the first time.  if the filter "reruns" (i.e. the behavior we dont want to see) on reprocessing, then the arguments will be swapped.

Reply:
I see, your view was to have it swap each time, if it turned out it didn't swap (i.e. swapped back), we know their is a bug.  I was thinking about it the opposite direction, only swap if there's a bug.

in your case, it will find a bug in the filter if the filter doesn't swap at all, in my case, if the filter doesn't swap at all, it wont find.  In either case, if the test is flakey and its reprocessed multiple times, it wont do it.

I'll do your idea as I understand, I see why its better now